pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -262,26 +263,42 @@ func (s *NodeController) DoChecks(nodes *api.NodeList) *api.NodeList {
 // DoCheck performs health checking for given node.
 func (s *NodeController) DoCheck(node *api.Node) []api.NodeCondition {
 	var conditions []api.NodeCondition
+
+	// Check Condition: NodeReady. TODO: More node conditions.
+	oldReadyCondition := s.getCondition(node, api.NodeReady)
+	newReadyCondition := s.checkNodeReady(node)
+	newReadyCondition.LastTransitionTime = util.Now()
```

Comment:
I haven't gave it much thought.  Sometimes I do use this pattern, i.e. "The default value will be x, unless condition c is true, which we then set the value to y".  This doesn't work well here I guess.